ddtrace/contrib/aws_lambda/patch.py

```diff
@@ -121,19 +121,27 @@ def _get_handler_and_module():
     path = os.environ.get("DD_LAMBDA_HANDLER", None)
     _datadog_instrumentation = DatadogInstrumentation()
 
+    handler, handler_module = None, None
+
     if path is None:
         from datadog_lambda.wrapper import datadog_lambda_wrapper
 
+        handler_module = datadog_lambda_wrapper
         handler = getattr(datadog_lambda_wrapper, "__call__")
-
-        return handler, datadog_lambda_wrapper, _datadog_instrumentation
     else:
         parts = path.rsplit(".", 1)
         (mod_name, handler_name) = parts
         modified_mod_name = _modify_module_name(mod_name)
         handler_module = import_module(modified_mod_name)
         handler = getattr(handler_module, handler_name)
-        return handler, handler_module, _datadog_instrumentation
+
+        # Check if handler is a class instance that can be called.
+        if not hasattr(handler, "__code__") and hasattr(handler, "__call__"):
```

Comment:
You're right. I'm wondering of another way of figuring out if the handler is a class. Do you happen to know? Since all instance classes are objects, I couldn't figure out another one. Also, this will never occur with our use cases for AWS Chalice, which don't create `__code__` in their classes. Another thing, users mostly use functions instead of classes for their AWS Lambda functions. What do you think?